x-pack/solutions/security/plugins/security_solution/server/lib/trial_companion/routes/nba_routes.ts

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  IKibanaResponse,
+  KibanaResponseFactory,
+  Logger,
+  RequestHandler,
+} from '@kbn/core/server';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { buildSiemResponse } from '@kbn/lists-plugin/server/routes';
+import type { TypeOf } from '@kbn/config-schema';
+import { schema } from '@kbn/config-schema';
+import { type Either, left, match, right } from 'fp-ts/Either';
+import { pipe } from 'fp-ts/function';
+import type { TrialCompanionUserNBAService } from '../services/trial_companion_user_nba_service.types';
+import { TRIAL_COMPANION_NBA_URL } from '../../../../common/trial_companion/constants';
+import type { SecuritySolutionRequestHandlerContext } from '../../../types';
+import { TrialCompanionUserNBAServiceImpl } from '../services/trial_companion_user_nba_service';
+import type { TrialCompanionRoutesDeps } from '../types';
+
+// TODO: think about dependency injection of saved objects (internal) client. Currently, we get it from context.core.savedObjects
+// TODO: if we need internal client - add user milestone service in x-pack/solutions/security/plugins/security_solution/server/request_context_factory.ts
+// TODO: and get it from request context
+
+interface NBAContext {
+  nbaService: TrialCompanionUserNBAService;
+  username: string;
+}
+
+const NBARouteSchema = {
+  body: schema.object({
+    milestoneId: schema.number(), // TODO: oneOf plus iterate over ALL_NBA keys
+  }),
+};
+
+type NBASeenRouteRequestBody = TypeOf<typeof NBARouteSchema.body>;
+
+const notFound: RequestHandler<never, never, never, SecuritySolutionRequestHandlerContext> = async (
+  context,
+  request,
+  response
+) => {
+  return response.notFound({
+    body: 'disabled',
+  });
+};
+
+export const registerGetNBARoute = ({ router, logger, enabled }: TrialCompanionRoutesDeps) => {
+  router.versioned
+    .get({
+      path: TRIAL_COMPANION_NBA_URL,
+      access: 'internal',
+      summary: 'Get Trial Companion NBA for a user',
+      options: {
+        tags: ['api'],
+      },
+      security: {
+        authz: {
+          requiredPrivileges: ['securitySolution'],
+        },
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: false, // TODO: help needed here - do we need to validate something?
+      },
+      enabled ? getCurrentNBAForUser(logger) : notFound
+    );
+};
+
+export const registerPostNBASeenRoute = ({ router, logger, enabled }: TrialCompanionRoutesDeps) => {
+  router.versioned
+    .post({
+      path: TRIAL_COMPANION_NBA_URL,
```

Comment:
nitpick: Maybe to make the API more "restful," we could define it slightly differently? e.g., `/internal/security_solution/trial_companion/nba/{id}`, it could use even a more accurate verb like PATCH instead of POST?

Reply:
hm... interesting. Lets think together. What is our "resource" here? `GET` method returns a "current" milestone, in other words there is no sense to create get by milestone id method. 

this post method is actually _user_ related, and not milestone resource related. In other words `POST` to the same URL `/internal/security_solution/trial_companion/nba/{id}` but with different users will results in different states. 

Do you think we could introduce REST API for this behaviour?